tests/__init__.py

```diff
@@ -0,0 +1,116 @@
+    "metadata",
+    "sqlData",
+    "stageData",
+    "taskData",
+]
+
+# In parsed log files (i.e. SparkApplication.to_dict()), this is the set of top-level keys for which we would expect
+# members to not necessarily always be ordered the same way
+UNORDERED_PARSED_KEYS = [
+    "accumData",
+    "sqlData",
+    "executors"
+]
+
+APP_NAME_INCORRECT_MESSAGE = "Application name does not match expected value"
+PARSED_KEYS_MISSING_MESSAGE = "Not all keys present in parsed eventlog"
+
+
+def assert_all_files_identical(parsed_files):
+    """
+    Given a list of already-parsed eventlog file dictionaries (i.e. SparkApplication.to_dict()), assert that all
+    dictionaries are meaningfully the same, ignoring ordering of keys as is applicable
+    """
+    [first, *rest] = parsed_files
+    for curr in rest:
+        for key, curr_value in curr.items():
+            first_value = first[key]
+
+            # This is much faster than a DeepDiff, and since we expect these values to match most of the time, this
+            #  should help speed up tests in the common-case that things are working as expected
+            if curr_value == first_value:
+                continue
+
+            diff = DeepDiff(curr_value, first_value)
+            # If we get a diff, we should double-check that that diff isn't just due to ordering differences, which may
+            #  not be an issue at all. This should help more readily pinpoint what the actual issue in any detected diff
+            #  may be
+            if diff:
+                diff_ignoring_order = DeepDiff(curr_value, first_value, ignore_order=True)
+
+            if diff and not diff_ignoring_order and key not in UNORDERED_PARSED_KEYS:
+                raise ValueError(f"Detected an ordering difference for key: {key} in parsed log files.\n" +
+                                 f"If this is expected, please update UNORDERED_PARSED_KEYS. Otherwise, please " +
+                                 "fix this diff.\n"
+                                 f"{pformat(diff)}")
+
+            if diff and diff_ignoring_order:
+                raise ValueError(f"Detected a difference not due to ordering for key: {key} in parsed log files\n" +
+                                 f"{pformat(diff_ignoring_order)}")
+
+
+def zip_to_tgz(zip_path: str | Path, file_suffix: str):
+    """
+    Given a zip archive residing at the provided Path, converts to a .tgz archive
+    """
+    zip_path = zip_path if isinstance(zip_path, Path) else Path(zip_path)
+    with tempfile.TemporaryDirectory() as temp_dir:
+        with zipfile.ZipFile(zip_path) as zfile:
+            zfile.extractall(temp_dir)
+            # Grab the last part of the original path to use as the filename for the new one
+            tgz_filename = zip_path.parts[-1]
+            # Remove suffixes from the original file name in reverse order
+            for suffix in zip_path.suffixes[::-1]:
+                tgz_filename = tgz_filename.removesuffix(suffix)
```

Comment:
Seems that `.stem` just removes `suffixes[-1]`, at least based on the example in the docs - 

```python
PurePath.stem
The final path component, without its suffix:

>>>
>>> PurePosixPath('my/library.tar.gz').stem
'library.tar'
>>> PurePosixPath('my/library.tar').stem
'library'
>>> PurePosixPath('my/library').stem
'library'
```

whereas I was going for removing _all_ of them. I could use `.stem` in a loop, though, I suppose. Not sure if that's any better...

Reply:
Ah, right. `pathlib` should make this easier. FWIW, here's a way to avoid the explicit iteration:
```
>>> p = Path("/tmp/something.tar.gz")
>>> p.name
'something.tar.gz'
>>> p.name.removesuffix("".join(p.suffixes))
'something'
```

Reply:
ah, nice - that's definitely cleaner!